apps/frontend/src/libs/components/sidebar/sidebar.tsx

```diff
@@ -1,40 +1,71 @@
-import logo from "~/assets/img/svg/auth-circle-mobile-logo.svg";
+import React from "react";
+
+import logo from "~/assets/img/svg/auth-circle-logo.svg";
 import { getValidClassNames } from "~/libs/helpers/helpers.js";
+import { useCallback, useState } from "~/libs/hooks/hooks.js";
 import { type MenuItem } from "~/libs/types/types.js";
 
+import { BlurredBackground } from "../blurred-background/blurred-background.js";
 import { Button } from "../button/button.js";
 import { Image } from "../image/image.js";
 import { Link } from "../link/link.js";
 import styles from "./styles.module.css";
 
 type Properties = {
-	className?: string | undefined;
 	menuItems: MenuItem[];
 };
 
-const Sidebar: React.FC<Properties> = ({
-	className,
-	menuItems,
-}: Properties) => {
+const Sidebar: React.FC<Properties> = ({ menuItems }: Properties) => {
+	const [isOpen, setOpen] = useState<boolean>(false);
+
+	const handleCloseSidebar = useCallback(() => {
+		setOpen(false);
+	}, []);
+
+	const toggleOpenSidebar = useCallback(() => {
+		setOpen(!isOpen);
+	}, [isOpen]);
+
 	return (
-		<div className={getValidClassNames(className, styles["wrapper"])}>
-			<Link className={styles["title-container"]} to="/">
-				<Image alt="website logo" className={styles["logo"]} src={logo} />
-				<h1 className={styles["title"]}>TrackMates</h1>
-			</Link>
-
-			<nav className={styles["menu"]}>
-				{menuItems.map(({ href, icon, label }) => (
-					<Button
-						className={styles["menu-item"]}
-						href={href}
-						iconName={icon}
-						key={label}
-						label={label}
-					/>
-				))}
-			</nav>
-		</div>
+		<>
+			<Button
+				className={getValidClassNames(
+					styles["burger-button"],
+					styles[isOpen ? "open" : "close"],
+				)}
+				hasVisuallyHiddenLabel
+				iconName="burger"
+				label="burger-button"
+				onClick={toggleOpenSidebar}
+			/>
+			<div
+				className={getValidClassNames(
+					styles["sidebar"],
+					styles[isOpen ? "open" : "close"],
+				)}
+			>
+				<Link className={styles["title-container"]} to="/">
+					<Image alt="website logo" className={styles["logo"]} src={logo} />
+					<h1 className={styles["title"]}>TrackMates</h1>
+				</Link>
+				<nav className={styles["menu"]}>
+					{menuItems.map(({ href, icon, label }) => (
+						<Button
+							className={styles["menu-item"]}
+							href={href}
+							iconName={icon}
+							key={label}
+							label={label}
+						/>
+					))}
+				</nav>
+			</div>
+			<BlurredBackground
+				className={styles["blurred-background"]}
+				isVisible={isOpen}
+				onClick={handleCloseSidebar}
```

Comment:
```suggestion
				onClick={handleToggleSidebar}
```
This component will be present only if `isOpen` is true, that means toggle will make it false